hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,570 @@
+            'state': shipping_address.state_id.name if shipping_address.state_id.name else '',
+            'zip': shipping_address.zip,
+            'country': shipping_address.country_id.name if shipping_address.country_id else ''
+        }
+
+        # Obtener los registros de envío relacionados
+        pickings = order.picking_ids
+
+        # Preparar datos de los envíos para la respuesta
+        shipping_data = []
+        for picking in pickings:
+            lines_data = [{
+                'product': line.product_id.name,
+                'quantity': line.product_uom_qty,
+                'done': line.quantity_done,
+                'name': line.name
+            } for line in picking.move_lines]
+
+            shipping_info = {
+                'name': picking.name,
+                'scheduled_date': picking.scheduled_date,
+                'state': picking.state,
+                'carrier': picking.carrier_id.name if picking.carrier_id else 'No definido',
+                'tracking_reference': picking.carrier_tracking_ref,
+                'lines': lines_data,
+                'address_data': address_data
+            }
+            shipping_data.append(shipping_info)   
+
+        shipping_data.append({
+            "sales_team": order.team_id.id,
+            "market_place_reference": order.channel_order_reference,
+            "sale_order_name": order.name,
+            "picking_id": picking.id,
+            "picking_name": picking.name
+        })     
+
+        return {"message": "Datos de envío obtenidos", "shipping_data": shipping_data}
+    
+    @route('/download_invoice/<int:invoice_id>', methods=['GET'], type='http', auth='user')
+    def download_invoice(self, invoice_id):
+        # Buscar factura
+        invoice = request.env['account.move'].browse(invoice_id)
+
+        # Generar el PDF
+        pdf_content, _ = request.env.ref('account.account_invoices')._render_qweb_pdf([invoice_id])
+
+        http_headers = [
+            ('Content-Type', 'application/pdf'),
+            ('Content-Length', len(pdf_content)),
+        ]
+        return request.make_response(pdf_content, headers=http_headers)
+    
+    @route('/stamp_invoice/<int:invoice_id>', methods=['POST'], auth='user', type='json')
+    def stamp_invoice(self, invoice_id=None):
+
+        # Buscar la factura por su ID
+        invoice = request.env['account.move'].browse(invoice_id)
+
+        # Verificar si la factura ya está timbrada
+        if invoice.l10n_mx_edi_cfdi_uuid:
+            return {"message": "La factura ya está timbrada.", "UUID": invoice.l10n_mx_edi_cfdi_uuid}
+        
+        # Intentar el proceso de timbrado
+        invoice.action_process_edi_web_services()
+        
+        # Verificar nuevamente si la factura fue timbrada exitosamente
+        if invoice.l10n_mx_edi_cfdi_uuid:
+            return {"success": "La factura ha sido timbrada con éxito.", "UUID": invoice.l10n_mx_edi_cfdi_uuid}
+        else:
```

Comment:
Unnecessary "else" after "return", remove the "else" and de-indent the code inside it